src/diagnostics.rs

```diff
@@ -20,7 +20,12 @@ pub fn report_diagnostic<'tcx, 'mir>(
             let info = info.downcast_ref::<TerminationInfo>().expect("invalid MachineStop payload");
             match info {
                 TerminationInfo::Exit(code) => return Some(*code),
-                TerminationInfo::Abort => format!("the evaluated program aborted execution"),
+                TerminationInfo::Abort(opt_msg) =>
+                    if let Some(msg) = opt_msg {
```

Comment:
You can fold the inner `if let` into the outer `match`:
```
TerminationInfo::Abort(Some(msg)) =>
TerminationInfo::Abort(None) =>
```

Reply:
Nice. forgot this was even a thing :)